### ISSUE TITLE
"Move to top" and "move to bottom" buttons for book list items

### ISSUE THREAD

**Issue by el4ctr0n** [internetarchive/openlibrary#9023]
Dear developers,
Currently, dragging and dropping book list items is the only way to manually order a massive list of books on a certain topic, with the books deemed essential ranked higher (in the upper half) and the books deemed optional ranked lower (in the bottom half).

To streamline the task described above, I am proposing that each book list item comes with buttons "Move to top" and "Move to bottom" that would move that book list item all the way up to the top of the list or all the way down to the bottom of the list.

I appreciate you considering my suggestion.

**Comment by Barbara0608** [comment]
Hey @cdrini! Can I work on this issue?

**Comment by mekarpeles** [comment]
@Barbara0608 if you'd like to try designing a simple mockup to share here we'd be happy to provide feedback and talk about next steps!

**Comment by omwagh28** [comment]
@cdrini i want work on this

**Comment by el4ctr0n** [comment]
Can there please be an update on this feature? Thank you in advance!

**Comment by cdrini** [comment]
Howdy, apologies for the delays folks! This is a good first issue if someone has time :+1:

Here's a mockup of how this might look:

![Image](https://github.com/user-attachments/assets/2a292d68-53e8-4282-b6df-05b3bc5b7200)

With HTML ( https://github.com/internetarchive/openlibrary/blob/2077bcae8438e374e0a01e3f5c130c57433cf12e/openlibrary/templates/type/list/edit.html#L26-L30 )

```html
<div class="seed--controls">
        <div class="mia__reorder mia__index ui-sortable-handle">≡ #1</div>
        <button class="mia__move" type="button">Move...</button>
        <button class="mia__remove" type="button">Remove</button>
    </div>
```

```css
.seed--controls {
    gap: 10px
}
```

Clicking "Move..." will bring up a JS `prompt` to set the index:

![Image](https://github.com/user-attachments/assets/3c10e531-2e9a-49df-861c-271f9671da86)

This lets us add fewer buttons, keeping the UX cleaner, and also gives folks a more powerful way to reorder items.

The JS edits will likely be near here in the code:

https://github.com/internetarchive/openlibrary/blob/2e387f5eaea7291b61ebdfdbb150096cdc4a3672/openlibrary/plugins/openlibrary/js/autocomplete.js#L218-L224

**Comment by cdrini** [comment]
@omwagh28 Apologies for the delay; if you are still interested this is actionable now :)

**Comment by pfolex** [comment]
I am interested as well.

**Comment by pfolex** [comment]
I have a branch with potential fix complete. 
https://github.com/pfolex/openlibrary#

I was not sure if I added the css update in the correct place. Please let me know if I can take this issue.

This code allows a user to click on move button, and then select a number to move to top, move to bottom, reposition, or cancel.

We could definitely discuss if we want to change this behavior.

**Comment by pfolex** [comment]
@cdrini Please see above.

**Comment by cdrini** [comment]
Awesome thanks @pfolex , assigning you! Noticing a few pieces in that code about "Move to top" buttons which don't exist, make sure to remove those. Otherwise looks like a great start, thank you for tackling, and please open a PR! :blush:

**Comment by pfolex** [comment]
This is what it looks like when you click on Move. I just wanted to make sure this is acceptable before creating a PR.

<img width="1411" alt="Image" src="https://github.com/user-attachments/assets/5f2c275b-4e37-4050-9634-29deff72d3c1" />

1. Click on the Move button.
2. Select 1 and ok to move the selection to the top.
3. Select 2 and ok to move the selection to the bottom.
4. Select 3 and ok to choose the position of the book in the list by number.
5. Select 4 to cancel moving the book within the list.

Thanks,

Alex

**Comment by cdrini** [comment]
Awesome thank you @pfolex !

Having the options is a bit too intricate, we want to keep it simple. The button is _only_ for moving to position. It should look like this: 

![Image](https://github.com/user-attachments/assets/3c10e531-2e9a-49df-861c-271f9671da86)

But I like the wording in yours, let's make it say: "Move to position (1-17):". Where 17 is the number of items on the list. That seems cleaner than using -1. This lets the user easily move to top, bottom, or anywhere else, with a single UI!

**Comment by pfolex** [comment]
I created a Pull Request.
https://github.com/internetarchive/openlibrary/pull/10936

**Comment by el4ctr0n** [comment]
Can you please add the "Move" button to the regular list view? Compared to the list editing mode, regular list view works much better on mobile devices.

Please see attached image for how this might look:

![Image](https://github.com/user-attachments/assets/38738279-cb25-4651-aecf-aee9265739c9)

**Comment by cdrini** [comment]
Unfortunately that would be outside the scope of this issue; the regular lists view doesn't have any of the code logic needed to be able to edit the list in this way. Let's finish the feature as currently defined for now!

Although I would be interested to hear what is cause trouble with the list editing experience on mobile @el4ctr0n .

**Comment by el4ctr0n** [comment]
@cdrini 
Here's how editing a large list looks on my phone: 

![Image](https://github.com/user-attachments/assets/a55f0251-c353-4368-b9f5-ed2bb4deba6d)
![Image](https://github.com/user-attachments/assets/53266b18-69d9-4c8b-9158-6a05ac24c40f)

As you see, the titles, author names, and cover images don't load.

**Comment by cdrini** [comment]
Ahhh that looks like a bug! I think it might have to do with the list being very long. Can you create a new issue for that?

**Comment by el4ctr0n** [comment]
@cdrini 
Yes, here it is: https://github.com/internetarchive/openlibrary/issues/10953
